crates/bevy_ecs/src/query/fetch.rs

```diff
@@ -712,6 +722,25 @@ unsafe impl<'a> QueryData for FilteredEntityRef<'a> {
         item
     }
 
+    #[inline]
+    fn provide_extra_access(
+        state: &mut Self::State,
+        access: &mut Access<ComponentId>,
+        available_access: &Access<ComponentId>,
+    ) {
+        // Claim any extra access that doesn't conflict with other subqueries
+        // This is used when constructing a `QueryLens` or creating a query from a `QueryBuilder`
+        // Start with the entire available access, since that is the most we can possibly access
+        state.clone_from(available_access);
+        // Prevent all writes, since `FilteredEntityRef` only performs read access
+        state.clear_writes();
+        // Prevent any access that would conflict with other accesses in the current query
+        state.remove_conflicting_access(access);
+        // Finally, add the resulting access to the query access
```

Comment:
This is interesting; because of how the macro is written, each term of a tuple inside a query like `Query<(&A, &B)>` sequentially updates the access by comparing with the access computed from the first `n` previous elements of the tuple?

Reply:
I think so?  It does two passes.  We first call `update_component_access` to determine the static access for things like `&A` and `&mut B`, and then call `provide_extra_access` to offer any leftover access.  So you can do either `(&mut A, FilteredEntityMut)` or `(FilteredEntityMut, &mut A)`, and the `&mut A` will claim the `A` access before the `FilteredEntityMut` sees it.  

But within `provide_extra_access`, it offers access to each term in order and looks at the access that previous terms took.  That causes the behavior you asked about below where `(FilteredEntityRef, FilteredEntityMut)` can be different from `(FilteredEntityMut, FilteredEntityRef)`.